include/ethernet.h

```diff
@@ -0,0 +1,92 @@
+/*
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ *  Copyright (C) 2021  The DOSBox Team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#ifndef DOSBOX_ETHERNET_H
+#define DOSBOX_ETHERNET_H
+
+#include <functional>
+
+#include "config.h"
+
+/** A virtual Ethernet connection
+ * While emulated Ethernet adapters provide the ability for the guest OS to
+ * send and receive Ethernet packets, the emulator itself needs to pass these
+ * packets to device on the host system.
+ * The EthernetConnection class provides a virtual connection for passing
+ * packets to and from a backend, intended for use by emulated adapter code.
+ */
+class EthernetConnection {
+public:
+	/** Initializes the connection.
+	 * After creating an EthernetConnection, it must have its connection
+	 * opened and initialized before further use.
+	 * This is done in a separate function instead of a constructor so any
+	 * error code may be propagated.
+	 * Do not call this function twice.
+	 * @param config The connection's configuration section
+	 * @return True if the connection is initialized, false otherwise
+	 */
+	virtual bool Initialize(Section *config) = 0;
+
+	/** Closes the connection.
+	 * Owners of an EthernetConnection are expected to delete it when
+	 * finished with the connection.
+	 * The destructor will close the connection and free any
+	 * per-connection resources.
+	 */
+	virtual ~EthernetConnection() {}
+
+	/** Sends a packet through the connection.
+	 * This function makes no guarantees as to whether the packet actually
+	 * gets stored or sent anywhere. Errors are not propagated from
+	 * the concrete implementation of the connection.
+	 * This matches the lossy nature of sending packets through hardware.
+	 * @param packet A pointer to bytes of an Ethernet frame
+	 * @param size The size (in bytes) of the Ethernet frame
+	 */
+	virtual void SendPacket(const uint8_t *packet, int size) = 0;
```

Comment:
Any reason for using `int` for packet size instead of `size_t`? (Not a problem, I'm just curious - using signed value for size might result in pointless casting later on, as stupid C/C++ integer promotion rules will need to be worked around).

Reply:
slirp uses int. Did you want me to convert it to a size_t?

Reply:
Library using `int` for this purpose is good enough reason to follow it in our code (we do the same in code calling SDL_net, which also uses `int`).